R/report.R

```diff
@@ -1,5 +1,5 @@
-Validator <- R6::R6Class(
-  classname = "Validator",
+Report <- R6::R6Class(
```

Comment:
The new class name surprised me. I think I got used to the previous one, but now I start to like `Report` :D

I'm curious why you decided to remove this: https://github.com/Appsilon/data.validator/pull/8/commits/f1c22fcf5acecbd1b425965e8b3b6e14a174330d#diff-a49a797e82cb8d4bb9dae1e29fcb4bc2e6ba5216cbe790ba38b6ef2095586a9cR6

This makes the usage of the package much easier. You don't even need to know that sth like `Report` class exist.

Are there any reasons for that?

Reply:
I am glad you like it :) What does "validator" even mean? For a begginer it can be super ambiguous that you actually don't have any validation functions thanks to "validator" but you accumulate results there. Weird. You accumulate results in the report, you can display it, etc.

Why I removed `getDefaultValidator()`: the reason is that it overcomplicates the implementation of our package. In every assert_* function we have to remember to pass this argument in the pipe chain (previously as `attr(data, "data-validator")` ) only to have invisible `add_results()` in the end. To me it is not worth the benefit of "not knowing that Report exists". In my opinion it is the opposite - a developer must exactly know that Report object exists, to build correct mental model how data.validator works (so it is not like a magic):

1. I create a Report object
2. I run validate(data) and assertion pipeline
3. I save the results to the report
4. I can do with the Report whatever I want (display it, analyse, reuse again, I can even have multiple independent reports at once)

I can imagine you can have different point of view on that. I would love to discuss it this week on a call :)

Reply:
Additional reason is that global variable like `.validator` in a package most probably will not pass CRAN checks. I think we had such issues with shiny.worker. Also I found some mentions like this: https://stackoverflow.com/questions/55604073/cran-policy-on-use-of-global-variables

Reply:
I understand and agree. This simplifies the implementation a lot.
I'm still thinking with the global report we'd simplify the api a lot but let's leave it for now.